### ISSUE TITLE
Problem with SMS node

### ISSUE THREAD

**Issue by node-red-italia** [zinen/node-red-contrib-huawei-router#8]
Hi Zinen, I have try a new version of collection of your nodes for testing the sms feature. Now I have an response error from the node:

Cannot read property 'transitional' of undefined

My router is B525s-65a.

Thanks for all. Happy new year

**Comment by zinen** [comment]
Hi there

Which version have you installed? And Which node-red version. 

If your router doesnt support sms feature then you will get fake respons back.

**Comment by node-red-italia** [comment]
Hi, I have a 1.2.7 version of node-red and 1.1.0 for your library.
My router support a sms (is possible send sms with Huawei UI).

I have used a Huawei lanhost e Huawei wanhost with Same error.

**Comment by zinen** [comment]
I havn't seen your error at  my tests and B525s-65a is on the tested device list of the API so I think it should work. 

So lets look into the lanhost node then. As it does not require any input . Could you extract the full error code?

Try importing the code below into your flow. Its got a catch node that should revival some more error info. Paste your error here.

```json
[{"id":"afe4b328228a23d5","type":"inject","z":"47a91361.5a922c","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":190,"y":3860,"wires":[["d4fff9bea78e8e27"]]},{"id":"a0e2c9382192269d","type":"debug","z":"47a91361.5a922c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":660,"y":3860,"wires":[]},{"id":"d4fff9bea78e8e27","type":"huawei-lanhosts","z":"47a91361.5a922c","name":"","x":430,"y":3860,"wires":[["a0e2c9382192269d"]]},{"id":"c17f141749ddda5d","type":"catch","z":"47a91361.5a922c","name":"","scope":["d4fff9bea78e8e27"],"uncaught":false,"x":420,"y":3940,"wires":[["83abfca9ac3e306c"]]},{"id":"83abfca9ac3e306c","type":"debug","z":"47a91361.5a922c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":660,"y":3940,"wires":[]}]
```

**Comment by node-red-italia** [comment]
Hi zinen, I have try your flow and I copy the message error:

{"_msgid":"c03e819a.b9184","payload":1641205911083,"topic":"","error":{"message":"Cannot read property 'transitional' of undefined","source":{"id":"d4fff9bea78e8e27","type":"huawei-lanhosts","count":1}}}

I have try on a Virtual Machine witch node-red 2.0.5 and the node huawei-lanhost works correctly.  I need to upgrade node-red and nodejs. 

The node huawei send sms return this error:

100005: Request format error

I have tried more format but no result.

Thanks a lot

**Comment by zinen** [comment]
Could be an older nodejs version. I'm not sure the requirements for the API to work but currently running nodejs version 14.18.

You found a bug! Update to version 1.1.1 of this node. It should be fixed now 0aec9609bdebb892ebabe64f94b935615643bd50.

Please reply back and close the issue if all is working after this.

**Comment by node-red-italia** [comment]
Perfect zinen, it's work send sms. I have tried with the phone number into a node, from msg.number and an array of phone number. Now it's works very well.

My compliments

Thanks a lot
